lib/data/submission.js

```diff
@@ -43,6 +43,14 @@ const submissionXmlToFieldStream = (fields, xml, includeStructuralAttrs = false,
   const stack = new SchemaStack(fields, true);
   let textBuffer = ''; // agglomerates text nodes that come as multiple events.
   const parser = new hparser.Parser({
+    onend: () => {
+      if (!stack.hasExited()) {
+        outStream.destroy(new Error('Stack has not exited yet'));
```

Comment:
Is there a test where we can see this error? How about updating the following test: "should not hang given malformed non-closing xml"? We could assert that the stream not only ends, but also errors.

Though I think I must be misunderstanding something about that test. How did that test pass before and not hang? Wouldn't the `SchemaStack` never exit in that case?

Reply:
I've updated the error text to `Stream ended before stack was exhausted.`, and updated the tests to assert on this error message.